src/main/java/mate/academy/springbootwebgreqit/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -31,16 +34,20 @@ public class ShoppingCartServiceImpl implements ShoppingCartService {
 
     @Transactional
     @Override
-    public ShoppingCartDto getShoppingCartForCurrentUser(Long userId) {
-        User user = userRepository.findById(userId)
-                .orElseThrow(() -> new IllegalArgumentException("User not found"));
-        ShoppingCart shoppingCart = user.getShoppingCart();
+    public ShoppingCartDto getShoppingCartForCurrentUser(Authentication authentication,
+                                                         Long shoppingCartId) {
+        User authUser = userRepository.findByEmail(authentication.getName())
+                .orElseThrow(() -> new EntityNotFoundException("User not found with email: "
+                        + authentication.getName()));
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(shoppingCartId)
+                .orElseThrow(() -> new EntityNotFoundException("Shopping cart not "
+                        + "found for user with id: " + shoppingCartId));
         if (shoppingCart == null) {
-            throw new EntityNotFoundException("Shopping cart not found for user with ID "
-                    + userId);
+            throw new EntityNotFoundException("Shopping cart not found for user with email: "
+                    + authUser.getEmail());
         }
         Hibernate.initialize(shoppingCart.getCartItems());
```

Comment:
Why do you need these all Hibernate.initialize?

Reply:
iused it to initialize data, but now i don't need it, cause i use @Transactional